Projects/Tweeter/logic.js

```diff
@@ -0,0 +1,142 @@
+const Tweeter = function()
+{
+    const posts = 
+        [
+            {
+                text: "First post!",
+                id: "p1",
+                comments: [
+                    { id: "c1", text: "First comment on first post!" },
+                    { id: "c2", text: "Second comment on first post!!" },
+                    { id: "c3", text: "Third comment on first post!!!" }
+                ]
+            },
+            {
+                text: "Aw man, I wanted to be first",
+                id: "p2",
+                comments: [
+                    { id: "c4", text: "Don't wory second poster, you'll be first one day." },
+                    { id: "c5", text: "Yeah, believe in yourself!" },
+                    { id: "c6", text: "Haha second place what a joke." }
+                ]
+            }
+        ]
+        let commentIdCounter = 0
+        let postIdCounter = posts.length
+        for(let i = 0;i<posts.length;i++)
+        {
+            commentIdCounter += posts[i].comments.length
+        }
+        console.log(postIdCounter)
+        console.log(commentIdCounter)
+
+        const getPosts = () =>
+        {
+            console.log(posts)
+            return posts
+        }
+
+        const addPost = function(text){
+            postIdCounter++
+            const post = {
+                id: "p" + postIdCounter,
+                text,
+                comments: []
+            }
+            posts.push(post)
+
+        }
+
+        const addComment = function(text,postID){
+            commentIdCounter++
+            const comment = {
+                id: "c" + commentIdCounter,
+                text
+            }
+            
+            for(let post of posts)
+            {
+                if(post.id === postID)
+                {
+                    post.comments.push(comment)
+                }
+            }
+
+        }
+
+        const removePost = function(postID)
+        {
+            
+            for(let i= 0;i<posts.length;i++)
```

Comment:
use the modern for loop like you used on line 57. You could do `for(let i in posts)`